models/Ion_auth_model.php

```diff
@@ -268,15 +259,26 @@ public function __construct()
 	 */
 	public function hash_password($password, $salt = FALSE, $use_sha1_override = FALSE)
 	{
-		if (empty($password))
+		// Check for empty password, or password containing null char
+		if (empty($password) || strpos($password, "\0") !== FALSE)
```

Comment:
maybe just do a !isset() here?  I'm not sure what all the null char check might be for here.

Reply:
Sorry, I should have explained a bit more.
See this comment on doc: http://php.net/manual/en/function.password-hash.php#118603

Reply:
Ah I didnt know that, thanks for the info!